app/models/payment.rb

```diff
@@ -0,0 +1,5 @@
+class Payment < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :category_payments
+  has_many :categories, through: :category_payments
+end
```

Comment:
- [ ] Payment `name`, `amount`, and category (at least one) are mandatory for creating a payment, kindly validations for them.

Reply:
fixed